compiler/src/dotty/tools/dotc/interpreter/Interpreter.scala

```diff
@@ -117,11 +117,15 @@ class Interpreter(implicit ctx: Context) {
       case Typed(expr, _) =>
         interpretTreeImpl(expr, env)
 
-      // Getting the underlying value of a value class. The value class is evaluated as its boxed representation
-      // as values in the interpreter are `Object`s. Therefore we just get it from the enviroment as is.
-      case Select(qualifier, _)
+      case Select(qualifier, name)
           if tree.symbol.owner.isValueClass && tree.symbol.is(ParamAccessor) && env.contains(qualifier.symbol) =>
-        env(qualifier.symbol)
+        val value = env(qualifier.symbol)
+        val clazz = value.getClass
+        if (clazz.getCanonicalName != tree.symbol.owner.showFullName) value // Already unboxed
+        else {
+          val method = getMethod(clazz, name, Nil)
+          interpreted(method.invoke(value))
```

Comment:
A minor irrelevant issue: the name `interpreted` takes me a little while to realise there's no interpretation.

Reply:
I will change the name